### PR TITLE
Correct microdox/v2 pins and organize common defs

### PR DESCRIPTION
**Pull request by timtraversy**
## Description

I flashed my v2 microdox a few days ago, but none of the switches were working. I looked at the firmware and realized that the `config.h` files were incorrectly configured. The config in the `microdox` folder defined the `v1` pins, and the `v2` config didn't override these values. The `info.json` files *are* correct, so I based the `config.h` updates of those.

I left all shared configuration in the parent `config.h`, and put version specific configs in the folders.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).
- But I only have a v2 microdox, so I can't test this for v1.

### CODE REVIEW

File: keyboards/boardsource/microdox/config.h [hunk 1]
```diff
@@ -19,24 +19,21 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "config_common.h"
 
+#define MANUFACTURER Boardsource    
+#define PRODUCT microdox    
+#define PRODUCT_ID 0x0412    
+#define VENDOR_ID 0xF7E0
+
```
Comment (zvecr) [inline keyboards/boardsource/microdox/config.h:26]:
```suggestion
```

File: keyboards/boardsource/microdox/v1/config.h [hunk 2]
```diff
@@ -3,18 +3,16 @@
 
 #pragma once
 
-/*
- * Feature disable options
- *  These options are also useful to firmware size reduction.
- */
+#define DEVICE_VER 0x0100
```
Comment (zvecr) [inline keyboards/boardsource/microdox/v1/config.h:6]:
```suggestion
```

File: keyboards/boardsource/microdox/v2/config.h [hunk 3]
```diff
@@ -1,8 +1,22 @@
 // Copyright 2022 jack (@waffle87)
 // SPDX-License-Identifier: GPL-2.0-or-later
+
 #pragma once
 
-#undef RGB_DI_PIN
+#define DEVICE_VER 0x0200
+
```
Comment (zvecr) [inline keyboards/boardsource/microdox/v2/config.h:7]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
These are now in info.json

**Comment by zvecr** [comment]
Just as an FYI, most of the changes in this PR are incorrect as they are duplicating content already configured within `info.json`. Matrix pins, encoders, etc.

**Comment by timtraversy** [comment]
Ahh I thought `info.json` was just for the API, and `config.h` was for compiling. Closing in favor of yours.

**Pull request closed without merging**